server/src/main/java/org/opensearch/index/store/remote/utils/cache/LRUCache.java

```diff
@@ -192,8 +196,10 @@ public void clear() {
             usage = 0L;
             activeUsage = 0L;
             lru.clear();
-            for (Node<K, V> node : data.values()) {
-                data.remove(node.key);
+            final Iterator<Node<K, V>> iterator = data.values().iterator();
+            while (iterator.hasNext()) {
+                Node<K, V> node = iterator.next();
+                iterator.remove();
```

Comment:
clear method of LRUCache was giving ConcurrentModificationException while calling it, since we were using a for each loop (which uses an iterator) but we were not removing via the iterator.

This method did not have any usage or tests till now, hence was missed out earlier.
Corrected the implementation and added tests for this method as well.